lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -790,21 +755,21 @@ Error PDBImporterTypeVisitor::visitKnownRecord(CVType &Record,
       // Get the proper LF_MFUNCTION.
       auto &MemberFunction = InProgressConcreteFunctionMemberTypes[FnTypeIndex];
       TypeIndex ReturnTypeIndex = MemberFunction.ReturnType;
-      auto ReferencedTypeFromModel = getModelTypeForIndex(ReturnTypeIndex);
-      if (not ReferencedTypeFromModel) {
+      auto ModelReturnType = makeModelTypeForIndex(ReturnTypeIndex);
+      if (ModelReturnType.empty()) {
         revng_log(DILogger,
                   "LF_MFUNCTION: Unknown return type "
                     << ReturnTypeIndex.getIndex());
         // Avoid function types that have incomplete type.
         return Error::success();
       }
 
-      auto NewType = makeTypeDefinition<CABIFunctionDefinition>();
-      auto TypeFunction = cast<CABIFunctionDefinition>(NewType.get());
-      TypeFunction->ABI() = Model->DefaultABI();
+      auto NewDefinition = makeTypeDefinition<CABIFunctionDefinition>();
+      auto &Prototype = *cast<CABIFunctionDefinition>(NewDefinition.get());
+      Prototype.ABI() = Model->DefaultABI();
 
-      QualifiedType TheReturnType(*ReferencedTypeFromModel, {});
-      TypeFunction->ReturnType() = TheReturnType;
+      if (!ModelReturnType.empty() && !ModelReturnType->isVoid())
+        Prototype.ReturnType() = std::move(ModelReturnType);
```

Comment:
New behavior? We're now stopping using `void` as return type?

Reply:
I'm pretty sure we discussed that at some point. You were the one pushing for no-type as the return value because that doesn't show up in the serialized model.

Context: https://mattermost.rev.ng/revng/pl/a8anpumy438c787xzs34n6cq6a

Reply:
Yes, yes, now I remember.

We should backlog (but not too far) to try to enforce that the return type cannot be `void` in `verify` and see what happens.

Reply:
Hm, I already did :) Hence changes like this.

Reply:
:enterprise: